packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1103 @@
+    large: 50
+  },
+  spacious: {
+    medium: 48, // table-row-height-medium-spacious
+    large: 60
+  }
+};
+
+type Scale = 'large' | 'medium';
+
+export class S2TableLayout<T> extends UNSTABLE_TableLayout<T> {
+  protected isStickyColumn(node: GridNode<T>): boolean {
+    return node.props.isSticky;
+  }
+
+  protected buildCollection(): LayoutNode[] {
+    let [header, body] = super.buildCollection();
+    let {children, layoutInfo} = body;
+    // TableLayout's buildCollection always sets the body width to the max width between the header width, but
+    // we want the body to be sticky and only as wide as the table so it is always in view if loading/empty
+    if (children?.length === 0) {
+      layoutInfo.rect.width = this.virtualizer.visibleRect.width - 80;
+    }
+
+    return [
+      header,
+      body
+    ];
+  }
 
+  protected buildLoader(node: Node<T>, x: number, y: number): LayoutNode {
+    let layoutNode = super.buildLoader(node, x, y);
+    let {layoutInfo} = layoutNode;
+    layoutInfo.allowOverflow = true;
+    layoutInfo.rect.width = this.virtualizer.visibleRect.width;
+    layoutInfo.isSticky = true;
+    return layoutNode;
+  }
+
+  protected buildBody(y: number): LayoutNode {
+    let layoutNode = super.buildBody(y);
+    let {children, layoutInfo} = layoutNode;
+    // Needs overflow for sticky loader
+    layoutInfo.allowOverflow = true;
+    // If loading or empty, we'll want the body to be sticky and centered
+    if (children?.length === 0) {
+      layoutInfo.rect = new Rect(40, 40, this.virtualizer.visibleRect.width - 80, this.virtualizer.visibleRect.height - 80);
+      layoutInfo.isSticky = true;
+    }
+
+    return {...layoutNode, layoutInfo};
+  }
+
+  protected buildRow(node: GridNode<T>, x: number, y: number): LayoutNode {
+    let layoutNode = super.buildRow(node, x, y);
+    layoutNode.layoutInfo.allowOverflow = true;
+    // Needs overflow for sticky selection/drag cells
+    return layoutNode;
+  }
+
+  protected buildTableHeader(): LayoutNode {
+    let layoutNode = super.buildTableHeader();
+    // Needs overflow for sticky selection/drag column
+    layoutNode.layoutInfo.allowOverflow = true;
+    return layoutNode;
+  }
+
+  protected buildColumn(node: GridNode<T>, x: number, y: number): LayoutNode {
+    let layoutNode = super.buildColumn(node, x, y);
+    // Needs overflow for the resize handle
+    layoutNode.layoutInfo.allowOverflow = true;
+    return layoutNode;
+  }
+}
 
 export function Table(props: TableProps) {
-  return <AriaTable {...props} />;
+  let {
+    UNSAFE_style,
+    UNSAFE_className,
+    isQuiet = false,
+    density = 'regular',
+    overflowMode = 'truncate',
+    styles,
+    loadingState,
+    onLoadMore,
+    onResize: propsOnResize,
+    onResizeStart: propsOnResizeStart,
+    onResizeEnd: propsOnResizeEnd,
+    onAction,
+    ...otherProps
+  } = props;
+
+  let scale = (useIsMobileDevice() ? 'large' : 'medium') as Scale;
```

Comment:
Does useIsMobileDevice match the media query we use for scale in S2? I think we changed it from v3...

Reply:
oh whoops I forgot that was the case, I'll bring back the S2 specific useIsMobileDevice that I had added before